File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        
               
                unordered_map<int,int>gzip;
                gzip[0]=1;
                int sum=0;
                int ans=0;
                for(int i=0;i<nums.size();i++){
                    
                            sum+=nums[i];
                            if(gzip.find(sum-k)!=gzip.end()){
                                    ans+=gzip[sum-k];
                            }
                            
                            gzip[sum]++;
                }
        
                return ans;     
           
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    
    
    def subarraySum(self, nums: List[int], k: int) -> int:
    
        
        dic={}
        total=0
        cnt=0
        dic[0]=1
        for item in nums:
            
            total+=item
            
            if total-k in dic:
                cnt+=dic[total-k]
            
            if total not in dic:
                    dic[total]=0
            
            dic[total]+=1
        
        return cnt
                
            